### PR TITLE
Multiple LEGA instances and fixing the database user

### PR DESCRIPTION
**Pull request by silverdaz**
Central EGA supports user authentication _per_ Local EGA instance.

This PR simulates it in the fake Central EGA. Users are linked into a subfolder for each LEGA instance they have access to.

Moreover, the database permissions issue seems fixed in this context.
The docker container for Postgres creates a database per user, and gives permissions to it.
This means that the `lega` _postgres user_ will be granted access to the `lega` _database_.
That suits us. However, permissions are not properly granted if we choose another user. To be fixed.

### CODE REVIEW

**Comment by silverdaz** [comment]
It is necessary to change the test script to call `boot.sh` instead of `generate.sh`

**Comment by silverdaz** [comment]
Oh, and when you fetch the users' public key, look at `.trace.cega` instead of `.trace`

File: docker/images/db/db.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
-DROP DATABASE IF EXISTS lega;
-CREATE DATABASE lega;
+-- DROP DATABASE IF EXISTS lega;
+-- CREATE DATABASE lega;
```
Comment (viklund) [inline docker/images/db/db.sql]: Don't comment remove.

File: docker/entrypoints/keys.sh [hunk 2]
```diff
@@ -37,5 +34,8 @@ KEYGRIP=$(/usr/local/bin/gpg -k --with-keygrip ega@nbis.se | awk '/Keygrip/{prin
 unset GPG_PASSPHRASE
 
 echo "Starting the gpg-agent proxy"
-#exec ega-socket-proxy '0.0.0.0:9010' /run/ega/S.gpg-agent.extra --certfile /etc/ega/ssl.cert --keyfile /etc/ega/ssl.key
-exec ega-socket-proxy '0.0.0.0:9010' /root/.gnupg/S.gpg-agent.extra --certfile /etc/ega/ssl.cert --keyfile /etc/ega/ssl.key
+#ega-socket-proxy '0.0.0.0:9010' /root/.gnupg/S.gpg-agent.extra --certfile /etc/ega/ssl.cert --keyfile /etc/ega/ssl.key &
```
Comment (viklund) [inline docker/entrypoints/keys.sh]: Don't comment, delete.

File: docker/ega.yml [hunk 3]
```diff
@@ -162,6 +181,9 @@ services:
...
 volumes:
-  inbox:
-  staging:
-  vault:
+  inbox_swe1:
+  staging_swe1:
+  vault_swe1:
+  inbox_fin1:
+  # staging_fin1:
+  # vault_fin1:
```
Comment (viklund) [inline docker/ega.yml]: Should the `...fin1` stuff be commented here, it seems setup further up.

**Comment by dtitov** [comment]
Codacy has some fair points on this PR, I suggest to follow them.

**Comment by silverdaz** [comment]
Ok. I’ll look at it tomorrow.

File: docker/ega.yml (hunk 3 above)
Comment (silverdaz) [reply docker/ega.yml]: There are the volumes. Not the containers.

**Comment by dtitov** [comment]
What about this: Codacy

File: docker/ega.yml (hunk 3 above)
Comment (viklund) [reply docker/ega.yml]: Yes, I know. That doesn't really answer the question though.

**Comment by silverdaz** [comment]
The warning can be ignored because the Codacy shell script tests do not see that the variable is used in a function that is sourced from another file.

**Pull request merged**